File: LC94_Binary Tree Inorder Traversal_Recursion.py
# -*- coding: utf-8 -*-
"""
Created on Wed Nov  4 13:48:40 2020

@author: Three
"""

# Definition for a binary tree node.
class TreeNode:
     def __init__(self, val=0, left=None, right=None):
         self.val = val
         self.left = left
         self.right = right
class Solution:
    def inorderTraversal(self, root):
        if root is None: return [] # 审题，要输出list不是int

        return self.inorderTraversal(root.left) + [root.val] + self.inorderTraversal(root.right)

root = TreeNode(2)
root.left = TreeNode(10)
root.right = TreeNode(24)

res = Solution()
print(res.inorderTraversal(root))

File: LC94_Binary Tree Inorder Traversal_Recursion.cpp
class Solution {
public:
    void inorder(TreeNode* root, vector<int>& res) { // 树的中序遍历
        if (!root) {
            return;
        }
        inorder(root->left, res);
        res.push_back(root->val);
        inorder(root->right, res);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        inorder(root, res);
        return res;
    }
};
